opensearch-operator/pkg/reconcilers/securityconfig.go

```diff
@@ -0,0 +1,213 @@
+		// Delete old job
+		r.logger.Info("Deleting old update job")
+		opts := client.DeleteOptions{}
+		// Add this so pods of the job are deleted as well, otherwise they would remain as orphaned pods
+		client.PropagationPolicy(metav1.DeletePropagationForeground).ApplyToDelete(&opts)
+		err = r.Delete(r.ctx, &job, &opts)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		// Make sure job is completely deleted (when r.Delete returns deletion sometimes is not yet complete)
+		_, err = r.ReconcileResource(&job, reconciler.StateAbsent)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+	r.logger.Info("Starting securityconfig update job")
+	job = r.job(jobName, namespace, checksum, adminCertName)
+	if err := ctrl.SetControllerReference(r.instance, &job, r.Client.Scheme()); err != nil {
+		return ctrl.Result{}, err
+	}
+	_, err = r.ReconcileResource(&job, reconciler.StateCreated)
+	return ctrl.Result{}, err
+}
+
+func checksum(data map[string][]byte) (string, error) {
+	hash := sha1.New()
+	keys := make([]string, 0, len(data))
+	for k := range data {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	for _, key := range keys {
+		_, err := hash.Write([]byte(key))
+		if err != nil {
+			return "", err
+		}
+		value := data[key]
+		_, err = hash.Write(value)
+		if err != nil {
+			return "", err
+		}
+	}
+	return base64.StdEncoding.EncodeToString(hash.Sum(nil)), nil
+}
+
+func (r *SecurityconfigReconciler) job(jobName string, namespace string, checksum string, adminCertName string) batchv1.Job {
+	dns := builders.DnsOfService(r.instance)
+	adminCert := "/certs/tls.crt"
+	adminKey := "/certs/tls.key"
+	caCert := "/certs/ca.crt"
+	var volumes []corev1.Volume
+	var volumeMounts []corev1.VolumeMount
+	volumes = append(volumes, corev1.Volume{
+		Name:         "securityconfig",
+		VolumeSource: corev1.VolumeSource{Secret: &corev1.SecretVolumeSource{SecretName: r.instance.Spec.Security.Config.SecurityconfigSecret.Name}},
+	})
+	volumeMounts = append(volumeMounts, corev1.VolumeMount{
+		Name:      "securityconfig",
+		MountPath: "/securityconfig",
+	})
+	volumes = append(volumes, corev1.Volume{
+		Name:         "admin-cert",
+		VolumeSource: corev1.VolumeSource{Secret: &corev1.SecretVolumeSource{SecretName: adminCertName}},
+	})
+	volumeMounts = append(volumeMounts, corev1.VolumeMount{
+		Name:      "admin-cert",
+		MountPath: "/certs",
+	})
+
+	arg := "ADMIN=/usr/share/opensearch/plugins/opensearch-security/tools/securityadmin.sh;" +
+		"chmod +x $ADMIN;" +
+		"count=0;" +
+		fmt.Sprintf("until $ADMIN -cacert %s -cert %s -key %s -cd /securityconfig/ -icl -nhnv -h %s.svc.cluster.local -p 9300 || (( count++ >= 20 )); do", caCert, adminCert, adminKey, dns) +
+		"  sleep 20; " +
+		"done"
+	annotations := map[string]string{
+		checksumAnnotation: checksum,
+	}
+	terminationGracePeriodSeconds := int64(5)
+	backoffLimit := int32(0)
+
+	return batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{Name: jobName, Namespace: namespace, Annotations: annotations},
```

Comment:
why dont we move it to `builders` package  ?

Reply:
Good idea. Done